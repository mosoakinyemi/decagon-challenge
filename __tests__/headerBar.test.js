import React from 'react';
import renderer from 'react-test-renderer';
import {Text} from 'react-native';
import HeaderBar from '../src/shared/header-bar';

jest.useFakeTimers();

describe('<HeaderBar/>', () => {
  it('HeaderBar component renders correctly', () => {
    const text = 'Incorrect Path';
    const instance = renderer.create(<HeaderBar title={text} />);
    expect(instance).toMatchSnapshot();
  });

  it('HeaderBar component renders correctly after an update', () => {
    const value1 = 'Hello';
    const value2 = 'Hi';

    const instance = renderer.create(<HeaderBar title={value1} />);
    const textInstance = instance.root.findByType(Text);
    expect(textInstance.props.children).toBe(value1);

    instance.update(<HeaderBar title={value2} />);
    expect(textInstance.props.children).toBe(value2);
  });
});
