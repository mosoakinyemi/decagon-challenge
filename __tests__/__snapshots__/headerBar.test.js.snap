// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderBar/> HeaderBar component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#d8d8e8",
      "elevation": 5,
      "flexDirection": "row",
      "height": 90.5,
      "justifyContent": "space-between",
      "paddingHorizontal": 30,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 3.84,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#2f3138",
        "fontSize": 29.5,
        "fontWeight": "bold",
        "lineHeight": 44.5,
      }
    }
  >
    Incorrect Path
  </Text>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 50,
        "opacity": 1,
        "width": 110,
      }
    }
  />
</View>
`;
